From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Tue, 30 Dec 2025 12:33:43 -0600
Subject: [PATCH] Add a converter to remove custom attribute

This custom attribute is no longer used but blows up logs every time it
loads in

diff --git a/ca/spottedleaf/dataconverter/minecraft/converters/attributes/ConverterAbstractOldAttributesRename.java b/ca/spottedleaf/dataconverter/minecraft/converters/attributes/ConverterAbstractOldAttributesRename.java
index 1a6d02df13f3461402fed429f7fbdb5acee877c4..7e63938b45c0756a289ace1d815cbf65ea7644ab 100644
--- a/ca/spottedleaf/dataconverter/minecraft/converters/attributes/ConverterAbstractOldAttributesRename.java
+++ b/ca/spottedleaf/dataconverter/minecraft/converters/attributes/ConverterAbstractOldAttributesRename.java
@@ -24,9 +24,23 @@ public final class ConverterAbstractOldAttributesRename {
                     return null;
                 }
 
+                // EMC start
+                java.util.List<Integer> toRemove = new java.util.ArrayList<>();
                 for (int i = 0, len = attributes.size(); i < len; ++i) {
+                    MapType mapData = attributes.getMap(i);
+                    if (mapData != null) {
+                        if ("generic.targetRange".equals(mapData.getString("Name"))) {
+                            toRemove.add(i);
+                            continue;
+                        }
+                    }
                     RenameHelper.renameString(attributes.getMap(i), "Name", renamer);
                 }
+                while (!toRemove.isEmpty()) {
+                    int i = toRemove.removeLast();
+                    attributes.remove(i);
+                }
+                // EMC end
 
                 return null;
             }
